docs/ember/components.md

```diff
@@ -101,21 +126,19 @@ export default class ArgsDisplay extends Component {
   {{/each}}
 </ul>
 ```
-
 ### Understanding `args`
 
-Now, looking at that bit of code, you might be wondering how it knows what the type of `this.args` is. In the `constructor` version, we explicitly _named_ the type of the `args` argument. Here, it seems to just work automatically. This works because the type definition for a Glimmer component looks roughly like this:
+Looking at that example above, Typescript knows what types `this.args` has, but how? In the `constructor` version, we explicitly _named_ the type of the `args` argument. Here, it seems to just work automatically. This works because the type definition for a Glimmer component looks roughly like this:
 
-```typescript
-export default class Component<Args extends {} = {}> {
-  readonly args: Args;
-
-  constructor(owner: unknown, args: Args);
-}
-```
+```typescript 
+ export default class Component {
+   args: Readonly<Args>;
+   constructor(owner: unknown, args: Args);
+ }
+ ```
 
 {% hint style="info" %}
-Not sure what’s up with `<Args>` _at all_? We highly recommend the [TypeScript Deep Dive](https://basarat.gitbooks.io/typescript/) book’s [chapter on generics ](https://basarat.gitbooks.io/typescript/docs/types/generics.html) to be quite helpful in understanding this part.
+Not sure what’s up with `<Args>` or `<S>` _at all_? We highly recommend the [TypeScript Deep Dive](https://basarat.gitbooks.io/typescript/) book’s [chapter on generics ](https://basarat.gitbooks.io/typescript/docs/types/generics.html) to be quite helpful in understanding this part.
```

Comment:
Let's just leave these as they were as well, despite the slight incoherence. I will do a follow-up PR later today after merging yours which updates *all* of this language (and possibly does some restructuring as well).

Reply:
All done and both sections are back in their original state. Thanks!